### ISSUE TITLE
Concurrency control

### ISSUE THREAD

**Issue by Chicagoan2016** [MarimerLLC/cslaforum#334]
Hi All,
I know this question is not specific to CSLA but I would love to hear what fellow CSLAers have to say about this, In the sample applications Rocky has a timestamp field in sql tables for concurrency control. I almost always use stored procedures for data access (and transactions in  those stored procs, with 'Read committed' isolation level).
I was wondering if I still need a time stamp field in my tables?

kind regards

**Comment by ajj7060** [comment]
Yes, I'd think you'd probably still need something like this.  Using stored procedures / transactions doesn't prevent the concurrency problems the timestamp is used to solve.  One simple case is that two users open the same business object for editing, one saves, then a few minutes later the other user saves.  If the first user changed property1, and the second user changes only property2, the second user effectively wipes out the first users change.  If that is ok, then you can not have the timestamp, but if you need to detect and prevent such things from happening, that's what the timestamp is meant to address.
